src/plugins/intel_gpu/src/kernel_selector/kernels/resample/resample_kernel_onnx.cpp

```diff
@@ -0,0 +1,173 @@
+
+    dispatchData.lws[0] = 1;
+    dispatchData.lws[1] = sub_group_size;
+    dispatchData.lws[2] = 1;
+
+    if (arg.outputs[0].GetLayout() == DataLayout::bs_fs_zyx_bsv32_fsv16 ||
+        arg.outputs[0].GetLayout() == DataLayout::bs_fs_zyx_bsv32_fsv32) {
+        dispatchData.lws[2] = GetOptimalDivisor(dispatchData.gws[2]);
+    }
+
+    return dispatchData;
+}
+
+KernelsPriority ResampleKernelOnnx::GetKernelsPriority(const Params& /*params*/, const optional_params& /*options*/) const {
+    return FORCE_PRIORITY_4;
+}
+
+bool ResampleKernelOnnx::Validate(const Params& p, const optional_params& o) const {
+    const resample_params& params = static_cast<const resample_params&>(p);
+    if (params.resampleType != ResampleType::LINEAR_ONNX)
+        return false;
+
+    if (!Parent::Validate(p, o))
+        return false;
+
+    if (p.GetType() != KernelType::RESAMPLE || o.GetType() != KernelType::RESAMPLE)
+        return false;
+
+    if (params.inputs.empty())
+        return false;
+
+    const auto& input = params.inputs[0];
+    const auto & output = params.outputs[0];
+    if ( input.Batch().v != output.Batch().v || input.Feature().v != output.Feature().v)
+        return false;
+
+    return true;
+}
+
+static bool IsThreeSpatialResample(const resample_params &params) {
+    const auto& input = params.inputs[0];
+    const auto & output = params.outputs[0];
+
+    if (input.Dimentions() == 5 && input.Z().v != output.Z().v)
+        return true;
+
+    return false;
+}
+
+JitConstants ResampleKernelOnnx::GetJitConstants(const resample_params &params) const {
+    auto jit = Parent::GetJitConstants(params);
+
+    auto opt_x_block_size = GetOptimalBlockSize(params);
+    if (params.outputs[0].X().v > 32 && opt_x_block_size == 1) {
+        opt_x_block_size = GetOptimalDivisor(params.outputs[0].X().v, 32);
+    }
+
+    jit.AddConstant(MakeJitConstant("OUTPUT_X_BLOCK_SIZE", opt_x_block_size));
+    jit.AddConstant(MakeJitConstant("X_BLOCKS", CeilDiv(params.outputs[0].X().v, opt_x_block_size)));
+    jit.AddConstant(MakeJitConstant("SUB_GROUP_SIZE", sub_group_size));
+
+    size_t vec_size = 0;
+    if (params.inputs[0].GetLayout() == DataLayout::fs_b_yx_fsv32) {
+        vec_size = 2;
+        jit.AddConstant(MakeJitConstant("FEATURE_SLICE_SIZE", 32));
+    } else {
+        vec_size = 1;
+        jit.AddConstant(MakeJitConstant("FEATURE_SLICE_SIZE", 16));
+    }
+    if (IsThreeSpatialResample(params))
+        jit.AddConstant(MakeJitConstant("THREE_SPATIAL_RESAMPLE", ""));
+
+    jit.AddConstant(MakeJitConstant("VEC_SIZE", vec_size));
+
+    if (!params.fused_ops.empty()) {
+        std::vector<std::string> idx_order;
+        idx_order = {"batch", "OF_ID", "oy", "ox"};
+
+        FusedOpsConfiguration conf = {"", idx_order, "res", GetAccumulatorType(params), 1};
+        jit.Merge(MakeFusedOpsJitConstants(params, {conf}));
+    }
+
+    return jit;
+}
+
+Datatype ResampleKernelOnnx::GetUnitType(const base_params& params) const {
```

Comment:
As I cen see unit type is not used in the kernel, so maybe overriding of this method doesn't really make sense.

Reply:
agree